SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp

```diff
@@ -702,47 +704,56 @@ void USpatialActorChannel::UpdateSpatialPosition()
 {
 	SCOPE_CYCLE_COUNTER(STAT_SpatialActorChannelUpdateSpatialPosition);
 
-	// PlayerController's and PlayerState's are a special case here. To ensure that they and their associated pawn are 
-	// handed between workers at the same time (which is not guaranteed), we ensure that we update the position component 
-	// of the PlayerController and PlayerState at the same time as the pawn.
+	// When we update an Actor's position, we want to update the position of all the children of this Actor.
+	// If this Actor is a PlayerController, we want to update all of its children and its possessed Pawn.
+	// That means if this Actor has an Owner or has a NetConnection and is NOT a PlayerController
+	// we want to defer updating position until we reach the highest parent.
+	if ((Actor->GetOwner() != nullptr || Actor->GetNetConnection() != nullptr) && !Actor->IsA<APlayerController>())
+	{
+		return;
+	}
+
+	// Check that there has been a sufficient amount of time since the last update.
+	if ((NetDriver->Time - TimeSinceLastPositionUpdate) < (1.0 / GetDefault<USpatialGDKSettings>()->PositionUpdateFrequency))
+	{
+		return;
+	}
 
-	// Check that it has moved sufficiently far to be updated
-	const float SpatialPositionThreshold = 100.0f * 100.0f; // 1m (100cm)
+	// Check that the Actor has moved sufficiently far to be updated
+	const float SpatialPositionThresholdSquared = FMath::Square(GetDefault<USpatialGDKSettings>()->PositionDistanceThreshold);
```

Comment:
Should this `moved sufficiently` be relative to the size of the object?

Reply:
Hmm, if we ever want to do a different threshold per object, that should *definitely* be exposed to the dev.
Should we add a map for thresholds per actor class to the settings?

Reply:
I think that can happen later but sure. I can make a ticket.